pkg/cloudprovider/aws/object_store.go

```diff
@@ -128,12 +128,16 @@ func (o *objectStore) GetObject(bucket string, key string) (io.ReadCloser, error
 	return res.Body, nil
 }
 
-func (o *objectStore) ListCommonPrefixes(bucket string, delimiter string) ([]string, error) {
+func (o *objectStore) ListCommonPrefixes(bucket string, delimiter string, prefix string) ([]string, error) {
 	req := &s3.ListObjectsV2Input{
 		Bucket:    &bucket,
 		Delimiter: &delimiter,
 	}
 
+	if prefix != "" {
+		req.Prefix = &prefix
+	}
+
 	var ret []string
 	err := o.s3.ListObjectsV2Pages(req, func(page *s3.ListObjectsV2Output, lastPage bool) bool {
 		for _, prefix := range page.CommonPrefixes {
```

Comment:
I also had to modify the line below:

```
val := *prefix.Prefix
ret = append(ret, val[0:strings.LastIndex(val, delimiter)])
```

It appears that AWS is forgiving in some cases if you have a `//` (i.e. our current scheme) and not forgiving in others (i.e. with your change).